rr_platform/src/ultrasonic_array/ultrasonic_array_to_pcl.cpp

```diff
@@ -0,0 +1,232 @@
+
+string readLine(boost::asio::serial_port &port) {
+    string line = "";
+    bool inLine = false;
+    while (true) {
+        char in;
+        try {
+            boost::asio::read(port, boost::asio::buffer(&in, 1));
+            if (in == '\n') {
+                return line;
+            }
+            line += in;
+        } catch (
+            boost::exception_detail::clone_impl <boost::exception_detail::error_info_injector<boost::system::system_error>> &err) {
+            ROS_ERROR("Error reading serial port.");
+            ROS_ERROR_STREAM(err.what());
+        }
+
+    }
+}
+
+vector<double> parseLine(string line) {
+  vector<double> dist;
+  vector<string> strs;
+  boost::split(strs, line, boost::is_any_of(","));
+
+  for (int i = 0; i < strs.size(); i++) {
+    dist.push_back(atof(strs[i].c_str())); //convert string to double
+  }
+
+  return dist;
+}
+
+void drawWall(pcl::PointCloud<pcl::PointXYZ> &cloud, pcl::PointXYZ center, float length, int numPoints) {
+  length = (length + 1) / 2.0; //@Note: adds one to ensure we get at least the numPoints desired
+  float step = length / (float) numPoints / 2.0;
+  float dist = step;
+  for (int i = 0; i < numPoints / 2; i++) {
+    //draw half the line and mirror
+    pcl::PointXYZ point1(center.x, dist, 0); //#TODO: might need to swap x and y
+    pcl::PointXYZ point2(center.x, -dist, 0); //#TODO: might need to swap x and y
+
+    cloud.push_back(point1);
+    cloud.push_back(point2);
+    dist = dist + step;
+  }
+
+}
+
+void drawSemiCircle(pcl::PointCloud<pcl::PointXYZ> &cloud, pcl::PointXYZ point, float radius, int numPoints) {
+  numPoints = (numPoints + 1) / 2; //@Note: adds one to ensure we get at least the numPoints desired
+  float angleStep = (pi / 2.0) / numPoints;
+
+  float angle = 0.0; //#TODO: may need to shift this start angle
+
+  for (int i = 0; i < numPoints ; i++) {
+    //draw a quarter of a cicle and mirror
+    float x = radius * cos(angle);
+    float y = radius * sin(angle);
+
+    cloud.push_back(pcl::PointXYZ(point.x + radius - x, y, 0));
+    cloud.push_back(pcl::PointXYZ(point.x + radius - x, -y, 0));
+
+    angle = angle + angleStep;
+  }
+
+}
+
+void drawCircle(pcl::PointCloud<pcl::PointXYZ> &cloud, pcl::PointXYZ point, double radius, int numPoints) {
+  double angleStep = (2.0 * pi) / numPoints;
+
+  double angle = 0; //start angle
+
+  for (int i = 0; i < numPoints; i++) {
+    //draw a circle
+    double x = radius * cos(angle);
+    double y = radius * sin(angle);
+
+    cloud.push_back(pcl::PointXYZ(point.x + radius + x, y, 0)); //point.x + radius is center of circle
+    angle = angle + angleStep;
+
+  }
+
+}
+
+
+
+ros::Publisher pub;
+string sensor_base_link;
+string sensor_link;
+float update_rate;
+string serial_port_name;
+float distance_clip;
+float length;
+float radius;
+int num_sensors;
+int num_points;
+int draw_method;
+int baud_rate;
+
+
+int main(int argc, char** argv) {
+	ros::init(argc, argv, "ultrasonic_array");
+
+	ros::NodeHandle nh;
```

Comment:
Check the formatting in this area, Indentation is inconsistent and it looks like it switches between tabs and spaces a few times.